### PR TITLE
[HADXV-62] Add a CLI for the fake intake

### PR DESCRIPTION
**Pull request by L3n41c**
### What does this PR do?

Add a CLI for the fake intake.

### Motivation

Ease the debugging of e2e tests.

### Additional Notes

Ex of use:
```bash
fakeintakectl --help
```
```
fakeintakectl is a CLI for interacting with fake intake servers.

Usage:
  fakeintakectl [command]

Available Commands:
  completion  Generate the autocompletion script for the specified shell
  filter      Filter logs or metrics
  flush       Flush the server and reset aggregators
  get
  help        Help about any command
  route-stats Get route stats

Flags:
  -h, --help         help for fakeintakectl
      --url string   fake intake server url

Use "fakeintakectl [command] --help" for more information about a command.
```

```bash
fakeintakectl --url http://internal-lenaic-eks-fakeintake-1048988617.us-east-1.elb.amazonaws.com route-stats
```
```
+-------------------+-------+
|       ROUTE       | COUNT |
+-------------------+-------+
| /api/v1/check_run |   542 |
| /api/v1/metadata  |    14 |
| /api/v2/logs      |  1095 |
| /api/v2/series    |   542 |
| /intake/          |    46 |
+-------------------+-------+
```

```bash
fakeintakectl --url http://internal-lenaic-eks-fakeintake-1048988617.us-east-1.elb.amazonaws.com get log-service names
```
```
agent
apps-http-client
apps-nginx-server
cluster-agent
dogstatsd
kube-proxy
redis
stress-ng
```

```bash
fakeintakectl --url http://internal-lenaic-eks-fakeintake-1048988617.us-east-1.elb.amazonaws.com filter logs --service agent
```
```
[
  {
    "message": "2023-12-06 14:52:13 UTC | TRACE | INFO | (pkg/trace/info/stats.go:91 in LogAndResetStats) | No data received",
    "status": "info",
    "timestamp": 1701874333582,
    "hostname": "i-0054118644b8ecfc4",
    "service": "agent",
    "ddsource": "agent",
    "ddtags": "filename:0.log,dirname:/var/log/pods/datadog_dda-linux-datadog-6frfx_ca4badce-510a-4402-bd7a-31bc3623e2d8/trace-agent,pod_phase:running,kube_qos:BestEffort,kube_namespace:datadog,image_name:gcr.io/datadoghq/agent,short_image:agent,kube_app_name:dda-linux-datadog,kube_app_instance:dda-linux,kube_app_component:agent,kube_container_name:trace-agent,kube_app_managed_by:Helm,kube_service:dda-linux-datadog,image_tag:7.49.0,image_id:gcr.io/datadoghq/agent@sha256:ba69302b2af6b2ac3372d76036527ccbb8fc9710e62d5381699e275080eaf49a,kube_ownerref_kind:daemonset,kube_daemon_set:dda-linux-datadog,git.repository_url:https://github.com/DataDog/datadog-agent,kube_ownerref_name:dda-linux-datadog,pod_name:dda-linux-datadog-6frfx,container_id:c55f1d1e514c7fc361bc032d6ecddd11a324f0c8fcc75ecaba87d11079d223a4,display_container_name:trace-agent_dda-linux-datadog-6frfx,container_name:trace-agent"
  },
  {
    "message": "2023-12-06 14:52:11 UTC | CORE | WARN | (pkg/collector/python/datadog_agent.go:131 in LogMessage) | disk:67cc0574430a16ba | (disk.py:136) | Unable to get disk metrics for /host/var/run/containerd/io.containerd.runtime.v2.task/k8s.io/cb2efc4ca4ed342261a31dfee6d54afa8ae9bc75ac695fe75e5680fbbf67de86/rootfs/host/proc/sys/fs/binfmt_misc: [Errno 40] Too many levels of symbolic links: '/host/var/run/containerd/io.containerd.runtime.v2.task/k8s.io/cb2efc4ca4ed342261a31dfee6d54afa8ae9bc75ac695fe75e5680fbbf67de86/rootfs/host/proc/sys/fs/binfmt_misc'. You can exclude this mountpoint in the settings if it is invalid.",
    "status": "info",
    "timestamp": 1701874332072,
    "hostname": "i-01d8952d17868da7f",
    "service": "agent",
    "ddsource": "agent",
    "ddtags": "filename:0.log,dirname:/var/log/pods/datadog_dda-linux-datadog-x97p4_d9807a31-2c73-421d-a443-8311828e243b/agent,kube_ownerref_kind:daemonset,kube_app_instance:dda-linux,kube_app_component:agent,kube_qos:BestEffort,short_image:agent,image_tag:7.49.0,kube_app_name:dda-linux-datadog,kube_service:dda-linux-datadog,kube_daemon_set:dda-linux-datadog,pod_phase:running,kube_namespace:datadog,kube_app_managed_by:Helm,kube_container_name:agent,image_name:gcr.io/datadoghq/agent,image_id:gcr.io/datadoghq/agent@sha256:ba69302b2af6b2ac3372d76036527ccbb8fc9710e62d5381699e275080eaf49a,git.repository_url:https://github.com/DataDog/datadog-agent,kube_ownerref_name:dda-linux-datadog,pod_name:dda-linux-datadog-x97p4,container_id:cb2efc4ca4ed342261a31dfee6d54afa8ae9bc75ac695fe75e5680fbbf67de86,display_container_name:agent_dda-linux-datadog-x97p4,container_name:agent"
  },
[…]
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: .gitlab/fakeintake.yml [hunk 1]
```diff
@@ -3,6 +3,26 @@
...
+
+test_fakeintake:
+  stage: source_test
+  rules:
+    - <<: *on_fakeintake_changes
+  needs: []
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["arch:amd64"]
+  script:
+    - inv fakeintake.test
```
Comment (pducolin) [inline .gitlab/fakeintake.yml:24]: **🥜 nitpick**
We do already run fakeintake tests, it is a module in `modules.py`

https://github.com/DataDog/datadog-agent/blob/main/tasks/modules.py#L153

The advantage of using common tests job is that it handles the results upload to CI Visibility, plus runs linter. 

Latest test runs: https://app.datadoghq.com/ci/test-runs?query=test_level%3Atest%20%40test.service%3Adatadog-agent%20%40git.branch%3Amain%20%40test.suite%3A%2Afakeintake%2A&citest_explorer_sort=time%2Cdesc&index=citest&mode=sliding&saved-view-id=2163063&start=1701342550011&end=1701947350011&paused=false

**Review by pducolin — APPROVED** [review]
Amazing !!

File: .gitlab/fakeintake.yml (hunk 1 above)
Comment (L3n41c) [reply .gitlab/fakeintake.yml:24]: Good point !!!
I didn’t realize that the fakeintake unit tests were already running in an already job.
In this case, I think it’s completely useless to create yet another job.

Let’s get rid of it:
```suggestion
```

File: .gitlab/fakeintake.yml [hunk 2]
```diff
@@ -3,6 +3,17 @@
...
+  stage: binary_build
+  rules:
+    - <<: *on_fakeintake_changes
+  needs: []
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["arch:amd64"]
+  script:
+    - inv fakeintake.build
+
+
```
Comment (L3n41c) [inline .gitlab/fakeintake.yml:16]:
```suggestion
```

**Pull request merged**